experimental/bits/simd_fixed_size.h

```diff
@@ -1331,6 +1340,56 @@ template <int _Np>
 	});
       }
 
+    // _S_gather {{{2
+    template <typename _Tp, typename _Up>
+      static inline _SimdMember<_Tp>
+      _S_gather(const _Up* __mem, const __int_for_sizeof_t<_Up>* __idx,
+		_TypeTag<_Tp>) noexcept
+      {
+	return _SimdMember<_Tp>::_S_generate([&](auto __meta) {
+	  return __meta._S_gather(__mem, &__idx[__meta._S_offset],
+				  _TypeTag<_Tp>());
+	});
+      }
+
+    // _S_gather {{{2
+    template <typename _Tp, typename _Up>
+      static inline _SimdMember<_Tp>
+      _S_gather(const _Up* __mem, const _SimdMember<_Tp>& __idx,
+		_TypeTag<_Tp>) noexcept
+      {
+	return _SimdMember<_Tp>::_S_generate([&](auto __meta) {
+	  return __meta._S_gather(
+	    __mem, __idx.template _M_tuple_at<__meta._S_offset>(),
+	    _TypeTag<_Tp>());
+	});
+      }
+
+    // _S_scatter {{{2
+    template <typename _Tp, typename _Up>
+      static inline void
+      _S_scatter(const _SimdMember<_Tp>& __v, _Up* __mem,
+		 const __int_for_sizeof_t<_Up>* __idx, _TypeTag<_Tp>) noexcept
+      {
+	__for_each(__v, [&](auto __meta, auto __native) {
+	  __meta._S_scatter(__native, __mem, &__idx[__meta._S_offset],
+			    _TypeTag<_Tp>());
+	});
+      }
+
+    // _S_scatter {{{2
+    template <typename _Tp, typename _Up>
+      static inline void
+      _S_scatter(const _SimdMember<_Tp>& __v, _Up* __mem,
+		 const _SimdMember<_Tp>& __idx, _TypeTag<_Tp>) noexcept
```

Comment:
Same type question as above. Did you want different types for `__v` and `__idx`?

Reply:
Yes, I try to support different in the following manner, but it seems that is not best way after reading our comments.
static inline _SimdMember<_Tp>
_S_gather(const _Up* __mem, const _SimdMember<__int_for_sizeof_t<_Tp>>& __idx,
		_TypeTag<_Tp>) noexcept